tasks/kmt.py

```diff
@@ -733,40 +963,19 @@ def build(
     domains = filter_target_domains(vms, infra, arch)
     cc = get_compiler(ctx, arch)
 
-    build_from_scratch = needs_build_from_scratch(ctx, paths, domains, full_rebuild)
-
-    if build_from_scratch:
-        build_dependencies(
-            ctx, arch, "test/new-e2e/system-probe/test-runner/files/system-probe-dependencies.json", "./", stack=stack
-        )
+    if len(domains) == 0:
+        raise Exit(f"no vms found from list {vms}. Run `inv -e kmt.status` to see all VMs in current stack")
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:** ‏Could be replaced by asserts